### PR TITLE
Upgrade Nav Native to v50.0.0, Maps SDK to 10.0.0-beta.20, and Common SDK to v12.0.0

### PR DESCRIPTION
**Pull request by LukasPaczos**
### Description

Upgrades Nav Native, Maps SDK, and Common SDK. Opening as a draft since this branch currently won't compile until we upgrade the Maps SDK as well.

### Changelog

```
<changelog>Swapped the `com.mapbox.bindgen.Expected` generic params from `<Value, Error>` to `<Error, Value>` to match platform conventions.</changelog>
<changelog>Exposed `com.mapbox.bindgen.Expected#fold`, `com.mapbox.bindgen.Expected#map`, and other mapping functions.</changelog>
<changelog>Removed `Expected<E, V> ExpectedFactory#createValue()` and replaced it with `Expected<E, None> ExpectedFactory#createNone()` which means that value can never be `null` and improves consumption from Kotlin.</changelog>
```

cc @mapbox/navigation-android

### CODE REVIEW

**Comment by LukasPaczos** [comment]
Pushed a snapshot of the Maps SDK so that we can start the review and testing process.

**Comment by Guardiola31337** [comment]
`unit-tests` are ❌ 

```
> Task :libnavigation-base:compileDebugUnitTestKotlin FAILED
e: /root/code/libnavigation-base/src/test/java/com/******/navigation/base/trip/model/roadobject/RoadObjectMapperTest.kt: (251, 13): Too many arguments for public constructor TunnelInfo(p0: String?) defined in com.******.navigator.TunnelInfo
```

https://github.com/mapbox/mapbox-navigation-android/blob/8b846eaa0bdf236d3b3fe1f1bd865e4f2f4c8df9/libnavigation-base/src/test/java/com/mapbox/navigation/base/trip/model/roadobject/RoadObjectMapperTest.kt#L251

Needs to be removed ☝️

**Comment by Guardiola31337** [comment]
`static-analysis` is also failing - `license` is failing

File: libnavui-maneuver/src/test/java/com/mapbox/navigation/ui/maneuver/TurnIconHelperTest.kt [hunk 1]
```diff
@@ -35,10 +35,10 @@ class TurnIconHelperTest {
         val mockType: String? = null
         val mockDegrees: Float? = null
         val mockModifier: String? = null
-        val mockDrivingSide: String? = null
+        val mocDRIVINGSide: String? = null
```
Comment (abhishek1508) [inline libnavui-maneuver/src/test/java/com/mapbox/navigation/ui/maneuver/TurnIconHelperTest.kt:38]: typo?
cc @LukasPaczos

Comment (LukasPaczos) [reply libnavui-maneuver/src/test/java/com/mapbox/navigation/ui/maneuver/TurnIconHelperTest.kt:38]: ah, `find and replace` issue, I'll revert

**Comment by LukasPaczos** [comment]
This only needs a Maps SDK update whenever the `10.0.0-beta.20` is available.

**Review by Guardiola31337 — APPROVED** [review]

File: libnavui-maps/src/test/java/com/mapbox/navigation/ui/maps/route/arrow/RouteArrowUtilsTest.kt [hunk 2]
```diff
@@ -221,7 +221,7 @@ class RouteArrowUtilsTest {
             (shaftSourceValueSlot.captured.contents as HashMap<String, Value>)["maxzoom"]!!.contents
         )
         assertEquals(
-            "{\"type\":\"FeatureCollection\",\"features\":[]}",
+            "null",
```
Comment (Guardiola31337) [inline libnavui-maps/src/test/java/com/mapbox/navigation/ui/maps/route/arrow/RouteArrowUtilsTest.kt:224]: cc'ing @cafesilencio for visibility

**Comment by Guardiola31337** [comment]
Going ahead and merging here in preparation for the upcoming release cc @RingerJK

**Pull request merged**